engine-tests/src/tests/xcc.rs

```diff
@@ -158,6 +159,90 @@ fn test_xcc_eth_gas_cost() {
     );
 }
 
+#[test]
+fn test_xcc_external_fund() {
+    // In this test we intentionally do not bridge wNEAR into the Engine.
+    // The purpose of the `fund_xcc_sub_account` functionality is to allow using
+    // the XCC feature in an Engine instance where there is no bridged wNEAR.
+
+    // Set up Engine contract
+    let aurora = deploy_evm();
+    let chain_id = AuroraRunner::default().chain_id;
+    let mut signer = test_utils::Signer::new(libsecp256k1::SecretKey::parse(&[0xab; 32]).unwrap());
+    let signer_address = test_utils::address_from_secret_key(&signer.secret_key);
+    let xcc_wasm_bytes = contract_bytes();
+    aurora
+        .user
+        .call(
+            aurora.contract.account_id(),
+            "factory_update",
+            &xcc_wasm_bytes,
+            near_sdk_sim::DEFAULT_GAS,
+            0,
+        )
+        .assert_success();
+    let wnear_account = deploy_wnear(&aurora);
+
+    // Fund XCC sub-account
+    let fund_amount = 5 * ONE_NEAR;
+    let fund_args = aurora_engine::xcc::FundXccArgs {
+        target: signer_address,
+        wnear_account_id: Some(wnear_account.account_id.as_str().parse().unwrap()),
+    };
+    aurora
+        .user
+        .call(
+            aurora.contract.account_id(),
+            "fund_xcc_sub_account",
+            &fund_args.try_to_vec().unwrap(),
+            near_sdk_sim::DEFAULT_GAS,
+            fund_amount,
+        )
+        .assert_success();
+
+    let sub_account_id = format!(
+        "{}.{}",
+        signer_address.encode(),
+        aurora.contract.account_id.as_str()
+    );
+    let sub_account = aurora
+        .user
+        .borrow_runtime()
+        .view_account(&sub_account_id)
+        .unwrap();
```

Comment:
Sorry forgot to mention that we shouldn't have unwrap on prod code.

Reply:
Even in tests?

Reply:
Thanks for the review @joshuajbouw ! I think panics in test code is acceptable, but I did remove a couple panics from the production code in https://github.com/aurora-is-near/aurora-engine/pull/735/commits/597114b7802261b8d8a0d70985c6bd73e1d3b970

Reply:
Oh sorry, I jumped the gun a bit given the PR on the other repo. Yeah here it is fine of course.